tests/bls_tests.rs

```diff
@@ -0,0 +1,262 @@
+    fn verify(&self) {
+        let pubkeys_result: Result<Vec<PublicKey>, _> = self
+            .input
+            .pubkeys
+            .iter()
+            .map(|x| PublicKey::from_bytes(&decode_hex_with_prefix(&x)))
+            .collect();
+        let pubkeys: Vec<PublicKey> = match pubkeys_result {
+            Ok(pk) => pk,
+            // error handling for infinity pub key
+            Err(_) => {
+                assert!(!self.output);
+                return;
+            }
+        };
+        let pubkeys_ref: Vec<&PublicKey> = pubkeys.iter().map(|x| x).collect();
+        let message_slice: &[u8] = &decode_hex_with_prefix(&self.input.message);
+        let signature: Signature =
+            match Signature::from_bytes(&decode_hex_with_prefix(&self.input.signature)) {
+                Ok(sk) => sk,
+                // error handling for zero signature
+                Err(_) => {
+                    assert!(!self.output);
+                    return;
+                }
+            };
+        let verify_result =
+            fast_aggregate_verify(pubkeys_ref.as_slice(), message_slice, &signature);
+        assert_eq!(verify_result, self.output)
+    }
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct VerifyInput {
+    pubkey: String,
+    message: String,
+    signature: String,
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct VerifyTestIO {
+    input: VerifyInput,
+    output: bool,
+}
+impl VerifyTestIO {
+    fn verify(&self) {
+        let pubkey: PublicKey =
+            match PublicKey::from_bytes(&decode_hex_with_prefix(&self.input.pubkey)) {
+                Ok(pk) => pk,
+                // error handling for infinity pub key
+                Err(_) => {
+                    assert!(!self.output);
+                    return;
+                }
+            };
+        let message_vec: Vec<u8> = decode_hex_with_prefix(&self.input.message);
+        let message: &[u8] = &message_vec;
+        let signature: Signature =
+            match Signature::from_bytes(&decode_hex_with_prefix(&self.input.signature)) {
+                Ok(sk) => sk,
+                // this is the case where the tampered signature cannot be read in as real signature
+                Err(_) => {
+                    assert!(!self.output);
+                    return;
+                }
+            };
+        let verify_result = signature.verify(&pubkey, message);
+        assert_eq!(verify_result, self.output)
+    }
+}
+#[test]
+fn test_fast_aggregate_verify() {
+    let entries =
+        glob("consensus-spec-tests/tests/general/phase0/bls/fast_aggregate_verify/small/**/*.yaml")
+            .expect("Failed to read glob pattern");
+    for entry in entries {
+        let path = entry.unwrap();
+        let file = File::open(path).expect("File does not exist");
+        let test_case: FastAggVerifyTestIO =
+            serde_yaml::from_reader(file).expect("Is not well-formatted yaml");
+        test_case.verify()
+    }
+}
```

Comment:
if we have this general pattern across the test repos, what do you think about making a generic driver function?

something like:

```rust
fn execute_test_case<T>(path_glob: TheGlobType) {
    let entries =
        glob(path_glob)
            .expect("Failed to read glob pattern");
    for entry in entries {
        let path = entry.unwrap();
        let file = File::open(path).expect("File does not exist");
        let test_case: T =
            serde_yaml::from_reader(file).expect("Is not well-formatted yaml");
        test_case.verify()
    }
}
```

it may make more sense to write another test format before deciding on the right abstraction here but we should be able to abstract over the general "load this test and then run it" mechanism to avoid a lot of code duplication